lib/airrecord/faraday_rate_limiter.rb

```diff
@@ -0,0 +1,60 @@
+require 'thread'
+
+module Airrecord
+  class FaradayRateLimiter < Faraday::Middleware
+    class << self
+      attr_accessor :requests
+    end
+
+    def initialize(app, requests_per_second: nil, sleeper: nil)
+      super(app)
+      @rps = requests_per_second
+      @sleeper = sleeper || ->(seconds) { sleep(seconds) }
+      @mutex = Mutex.new
+      clear
+    end
+
+    def call(env)
+      @mutex.synchronize do
+        wait if too_many_requests_in_last_second?
+        @app.call(env).on_complete do |_response_env|
+          requests << Process.clock_gettime(Process::CLOCK_MONOTONIC)
```

Comment:
Will `Process.clock_gettime` work on Windows systems? I'm not set up to test it, but I don't see support listed in the Ruby docs. `Time.now.to_f` should have enough precision and might be a safer choice.

Reply:
Excellent point. I've only written software for Linux systems for years and default to monotonic clocks for _any_ increments. Force of habit; not necessary here.